### ISSUE TITLE
AWS Route 53 Records delete every apply

### ISSUE THREAD

**Issue by wazoo** [hashicorp/terraform#1264]
For some reason every time I run terraform apply it deletes all of the DNS records in the config and re-applies them, this creates a lot of blocking tasks.  

My records aren't anything complicated, here is an example:

```
resource "aws_route53_record" "app_stage" {
   zone_id = "${var.aws_hosed_zones.testing_internal}"
   name = "app-stg"
   type = "CNAME"
   ttl = "120"
   records = ["${aws_elb.lb_app.dns_name}"]
}
```
And this gives me (sanitized):
```
-/+ aws_route53_record.app_stage
    name:              "app_stage.domain.com" => "app_stage" (forces new resource)
    records.#:         "1" => "1"
    records.12345678: "app_stage-123456789.us-west-2.elb.amazonaws.com" => "app_stage-123456789.us-west-2.elb.amazonaws.com"
    ttl:               "120" => "120"
    type:              "CNAME" => "CNAME"
    zone_id:           "<zone>" => "<zone>"
```
Is this expected behavior?

**Comment by catsby** [comment]
Hey @wazoo – if you're not changing anything, then no this is not expected. Is this on 0.3.7?

**Comment by wazoo** [comment]
Yep.

```
$ terraform -v
Terraform v0.3.7
```

I don't know if it matters but the zone is an "internal" zone that is attached to a VPC and it is not managed by terraform (because I can't do private zones with TF right now).

**Comment by radeksimko** [comment]
@catsby This is really happening if you use the "auto-magic" feature from #969 
See https://github.com/hashicorp/terraform/issues/1122

It is possible to work around that, taking the example above:

```ruby
resource "aws_route53_record" "app_stage" {
   zone_id = "${var.aws_hosed_zones.testing_internal}"
   name = "app-stg.domain.com" # Notice the `.domain.com` here
   type = "CNAME"
   ttl = "120"
   records = ["${aws_elb.lb_app.dns_name}"]
}
```

**Comment by catsby** [comment]
Thanks @wazoo and @radeksimko ; 969 was well intended and a good idea I think, but so far it seems to be introducing more problems than the convenience is worth. I'm going to tinker with something this morning and discuss internally.

**Comment by wazoo** [comment]
Interesting, I see where you were going with 969, makes sense.  I guess my issue here was that I can't manage the zone with TF yet (its internal only).  I will use the FQDN for now, thats a pretty easy workaround.

**Comment by catsby** [comment]
The core issue here is fixed with #1279.  Route 53 records still need a proper update function, as detailed in #1122
